File: app/src/main/java/com/andylai/donutbuttons/MainActivity.java
package com.andylai.donutbuttons;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.res.ResourcesCompat;

import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		DonutButtonsView view = findViewById(R.id.donutButtonView);
		view.setDefaultIndex(1);
		view.setCallback(new DonutButtonsView.Callback() {
			@Override
			public void onCenterTouched() {
				Log.d("Andy", "onCenterTouched");
			}

			@Override
			public void onCircularButtonTouched(DonutButtonsView self, int index) {
				if (index == 0 || index == 1 || index == 2 || index == 8)
					self.setPressed(index);
			}
		});
		List<Drawable> icons = new ArrayList<>();
		icons.add(getCompatDrawable(R.drawable.ic_donut_eraser));
		icons.add(getCompatDrawable(R.drawable.ic_donut_pen));
		icons.add(getCompatDrawable(R.drawable.ic_donut_brush));
		icons.add(getCompatDrawable(R.drawable.ic_donut_close));
		icons.add(getCompatDrawable(R.drawable.ic_donut_delete));
		icons.add(getCompatDrawable(R.drawable.ic_donut_undo));
		icons.add(getCompatDrawable(R.drawable.ic_donut_redo));
		icons.add(getCompatDrawable(R.drawable.ic_donut_save));
		view.setIconDrawables(icons);
	}

	private Drawable getCompatDrawable(int resId) {
		return ResourcesCompat.getDrawable(getResources(), resId, null);
	}
}

File: app/src/main/java/com/andylai/donutbuttons/MainActivity.kt
package com.andylai.donutbuttons

import android.graphics.PointF
import android.os.Bundle
import android.util.Log
import android.view.MotionEvent
import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val circleAnnotationBar = CircleAnnotationBar(findViewById(R.id.circleAnnotationBar))

        circleAnnotationBar.setRedoEnabled(false)
        circleAnnotationBar.setUndoEnabled(false)

        findViewById<Button>(R.id.open).setOnClickListener {
            circleAnnotationBar.setOpen(true)
        }

        findViewById<Button>(R.id.close).setOnClickListener {
            circleAnnotationBar.setOpen(false)
        }

        findViewById<Button>(R.id.penHide).setOnClickListener {
            circleAnnotationBar.setPenVisibility(View.GONE)
        }

        findViewById<Button>(R.id.penShow).setOnClickListener {
            circleAnnotationBar.setPenVisibility(View.VISIBLE)
        }

        findViewById<Button>(R.id.brushHide).setOnClickListener {
            circleAnnotationBar.setBrushVisibility(View.GONE)
        }

        findViewById<Button>(R.id.brushShow).setOnClickListener {
            circleAnnotationBar.setBrushVisibility(View.VISIBLE)
        }
    }
}